python/src/otel/otel_sdk/opentelemetry/instrumentation/aws_lambda/__init__.py

```diff
@@ -189,37 +198,35 @@ def _instrumented_lambda_handler_call(call_wrapped, instance, args, kwargs):
             [wrapped_module_name, wrapped_function_name]
         )
 
-        lambda_event = args[0]
+        lambda_event, lambda_context = args[0], args[1]
 
         parent_context = _determine_parent_context(lambda_event)
 
+        if tracer_provider:
+            tracer = get_tracer(__name__, __version__, tracer_provider)
+        else:
+            tracer = get_tracer(
+                __name__,
+                __version__,
+                context_to_tracer_provider(lambda_context),
+            )
+
         with tracer.start_as_current_span(
             name=orig_handler_name, context=parent_context, kind=SpanKind.SERVER
         ) as span:
             if span.is_recording():
                 lambda_context = args[1]
-                # Refer: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/faas.md#example
+                # Refer: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/faas.md#example
                 span.set_attribute(
                     SpanAttributes.FAAS_EXECUTION, lambda_context.aws_request_id
                 )
-                span.set_attribute(
-                    "faas.id", lambda_context.invoked_function_arn
-                )
```

Comment:
This attribute is what motivated the large changes in this PR. Semantic conventions tells us this should be a `ResourceAttribute` and not a `SpanAttribute`. `Resource` can only be set once, and the `lambda_context` isn't available until later. That's why we have to do all this.

I wonder if this will make the Lambda take much longer if we can't initialize the `TracerProvider` before the Lambda gets invoked. My gut reaction says it won't take too much longer.